test/export/test_type_utils.py

```diff
@@ -0,0 +1,85 @@
+from typing import List, Union, Dict
+from torch._export.serde.type_utils import check, Succeed, Fail, NotSure
+from torch.testing._internal.common_utils import (
+    run_tests,
+    TestCase,
+)
+
+
+class JustATest:
+    def __repr__(self) -> str:
+        return "JustATest()"
+
+
+class JustAnotherTest:
+    def __repr__(self) -> str:
+        return "JustAnotherTest()"
+
+
+class TestUnionTypeCheck(TestCase):
```

Comment:
Could we format some of these tests with some sort of [`make_test`](https://github.com/pytorch/pytorch/blob/cebc11ae8fbc13de4eadee19c579712e5d282754/test/inductor/test_compiled_optimizers.py#L57).

So that you can just do:

```
make_test([[42, 120], [10, 5]], List[List[int]]), Succeed)
make_test(["yada", "hurray"], List[int], Fail)
```